researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -579,6 +590,24 @@ public void uploadTaskResult(Context context, @NonNull TaskResult taskResult) {
             }
         }
 
+        if (lastLoadedTaskGuid == null) {
+            logger.error("lastLoadedTaskGuid must be set for this task to complete and " +
+                    "set finishedOn and startedOn dates, and clientData on the bridge server");
+        } else {
+            // TODO: make GUID available in ScheduledActivity constructor and get rid of this gson nonsense
+            String activityJson = String.format("{\"guid\":\"%s\"}", lastLoadedTaskGuid);
+            ScheduledActivity activity = (new Gson()).fromJson(activityJson, ScheduledActivity.class);
+            if (taskResult.getStartDate() != null) {
+                activity.setStartedOn(new DateTime(taskResult.getStartDate()));
+            }
+            if (taskResult.getEndDate() != null) {
+                activity.setFinishedOn(new DateTime(taskResult.getEndDate()));
+            }
+            bridgeManagerProvider.getActivityManager().updateActivity(activity).subscribe(message -> {
+                logger.info("Update activity success " + message);
+            }, throwable -> logger.error(throwable.getLocalizedMessage()));
+        }
```

Comment:
Is it worth adding unit tests for this?

Reply:
I can do that, but how would I check to see if bridgeManagerProvider.getActivityManager().updateActivity gets called in the test?

Reply:
verify(mockActivityManager).updateActivity(eq(expectedActivity))

Reply:
Thanks!  I added that test